lib/src/core/implementation/signature_body_source_builder.dart

```diff
@@ -0,0 +1,99 @@
+///   uri: Uri.parse('https://example.com/path?query=string'),
+///   method: 'GET',
+///   body: utf8.encode('example body'),
+///   timestamp: DateTime.now(),
+///   nonce: 'unique_nonce',
+///   headers: {'Content-Type': 'application/json'},
+///   signatureBodySourceComponents: ['Method', 'Host', 'QueryString', 'Body'],
+/// );
+/// final signatureBodySource = await builder.build(parameters);
+/// print(signatureBodySource);
+/// ```
+class SignatureBodySourceBuilder implements ISignatureBodySourceBuilder {
+  @override
+  Future<Uint8List> build(SignatureBodySourceParameters parameters) async {
+    final result = <int>[];
+
+    for (var component in parameters.signatureBodySourceComponents) {
+      switch (component) {
+        case 'Method':
+          result.addAll(utf8.encode(parameters.method.toUpperCase()));
+          break;
+        case 'Scheme':
+          if (parameters.uri.isAbsolute) {
+            result.addAll(utf8.encode(parameters.uri.scheme));
+          }
+          break;
+        case 'Host':
+          if (parameters.uri.isAbsolute) {
+            result.addAll(utf8.encode(parameters.uri.host));
+          }
+          break;
+        case 'Port':
+          if (parameters.uri.isAbsolute) {
+            result.addAll(utf8.encode(parameters.uri.port.toString()));
+          }
+          break;
+        case 'LocalPath':
+          result.addAll(utf8.encode(parameters.uri.path));
+          break;
+        case 'QueryString':
+          result.addAll(utf8.encode(parameters.uri.query));
+          break;
+        case 'Body':
+          // Flatten the list of Uint8List into a single list of int
+          result.addAll(parameters.body ?? Uint8List(0));
+          break;
+        case 'Timestamp':
+          result.addAll(utf8.encode(parameters.timestamp.toString()));
+          break;
+        case 'Nonce':
+          result.addAll(utf8.encode(parameters.nonce));
+          break;
+        default:
+          if (SignatureBodySourceComponents.isHeader(component, null)) {
+            final headerName = SignatureBodySourceComponents.header(component);
+            print('Processing header component: $headerName');
+            if (parameters.headers.containsKey(headerName)) {
+              result.addAll(utf8.encode(parameters.headers[headerName]!));
+              print(
+                  'Header value found for $headerName: ${parameters.headers[headerName]}');
+            } else {
+              print('Header value not found for $headerName');
+              print('Headers available: ${parameters.headers}');
+            }
+          }
+      }
+    }
+
+    print('--Building signature body source:');
+    print('----Method: ${parameters.method}');
+    print('----URI: ${parameters.uri}');
+    print('----Headers: ${parameters.headers}');
+    print('----Nonce: ${parameters.nonce}');
+    print('----Timestamp: ${parameters.timestamp}');
+    print('----Client ID: ${parameters.clientId}');
```

Comment:
I don't think these prints should be there. They aren't there in the .NET implementation.
Also the Client ID should most probably not be printed for security reasons.